### PR TITLE
Support BigQuery DATE type

### PR DESCRIPTION
**Pull request by peihe**

### CODE REVIEW

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryAvroUtils.java [hunk 1]
```diff
@@ -205,6 +209,9 @@ private static Object convertRequiredField(
         byte[] bytes = new byte[byteBuffer.limit()];
         byteBuffer.get(bytes);
         return BaseEncoding.base64().encode(bytes);
+      case "DATE":
+        verify(v instanceof Utf8, "Expected Utf8, got %s", v.getClass());
```
Comment (tgroh) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryAvroUtils.java]: why is this not `CharSequence` (as done for strings)?

If it is expected to be, then this should be a shared case

Comment (peihe) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryAvroUtils.java]: You are right.
Merged to STRING case.

**Comment by peihe** [comment]
R: @lukecwik @dhalperi

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryTableRowIteratorTest.java [hunk 2]
```diff
@@ -168,10 +169,10 @@ public void testReadFromQuery() throws IOException, InterruptedException {
     String photoBytesEncoded = BaseEncoding.base64().encode(photoBytes);
     // Mock table data fetch.
     when(mockTabledataList.execute())
-        .thenReturn(rawDataList(rawRow("Arthur", 42, photoBytesEncoded)));
+        .thenReturn(rawDataList(rawRow("Arthur", 42, photoBytesEncoded, "2000-01-01")));
 
     // Run query and verify
-    String query = "SELECT name, count, photoBytes from table";
+    String query = "SELECT name, count, photoBytes, anniversary from table";
```
Comment (lukecwik) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryTableRowIteratorTest.java]: shouldn't photoBytes be "photo"

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryAvroUtilsTest.java [hunk 3]
```diff
@@ -83,7 +85,7 @@ public void testConvertGenericRecordToTableRow() throws Exception {
       assertEquals(row, convertedRow);
     }
     {
-      // Test type conversion for TIMESTAMP, INTEGER, BOOLEAN, BYTES, and FLOAT.
+      // Test type conversion for INTEGER, FLOAT, TIMESTAMP, BOOLEAN, BYTES, and DATE.
       GenericRecord record = new GenericData.Record(avroSchema);
       byte[] soundBytes = "chirp,chirp".getBytes();
       ByteBuffer soundByteBuffer = ByteBuffer.allocate(soundBytes.length).put(soundBytes);
```
Comment (lukecwik) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryAvroUtilsTest.java]: ByteBuffer.wrap(soundBytes)

**Comment by lukecwik** [comment]
Minor comments and then LGTM

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryTableRowIteratorTest.java (hunk 2 above)
Comment (peihe) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryTableRowIteratorTest.java]: done

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryAvroUtilsTest.java (hunk 3 above)
Comment (peihe) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryAvroUtilsTest.java]: done

**Pull request closed without merging**